src/MemoryMgmt.chpl

```diff
@@ -0,0 +1,140 @@
+            }
+        }
+
+        return (AutoMath.round(availableMemoryPct/100 * memAvail)*1000):uint(64);
+    }
+
+    proc localeMemAvailable(reqMemory) : bool throws {
+        var arkMemAlloc = getArkoudaMemAlloc();
+        var arkMemUsed = memoryUsed();
+        var availMemory = getAvailMemory();
+
+        mmLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
+                 "locale: %s reqMemory: %i arkMemAlloc: %i arkMemUsed: %i availMemory: %i".format(here.id,
+                                                                                                  reqMemory,
+                                                                                                  arkMemAlloc,
+                                                                                                  arkMemUsed,
+                                                                                                  availMemory));
+        var newArkoudaMemory = reqMemory:int + arkMemUsed:int;
+        
+        if newArkoudaMemory:int <= arkMemAlloc:int {
+            return true;
+        } else {
+            if newArkoudaMemory:int <= availMemory {
+                return true;
+            } else {
+                var msg = "Arkouda memory request %i on locale %s exceeds available memory %i".format(newArkoudaMemory,
+                                                                                                      here.id,
+                                                                                                      availMemory);
+                mmLogger.error(getModuleName(),getRoutineName(),getLineNumber(),msg);
+                return false;
+            }
+        }
+    }
+    
+    /*
+     * Returns a boolean indicating whether there is either sufficient memory within the
+     * memory allocated to Arkouda on each locale host; if true for all locales, returns true. 
+     * 
+     * If the reqMemory exceeds the memory currently allocated to at least one locale, each locale 
+     * host is checked to see if there is memory available to allocate more memory to each
+     * corresponding locale. If there is insufficient memory available on at least one locale,
+     * returns false. If there is sufficient memory on all locales to allocate sufficient, 
+     * additional memory to Arkouda to execute the command, returns true.
+     */
+    proc isMemAvailable(reqMemory) : bool throws {
+      var overMemLimit : bool = false;
+
+      coforall loc in Locales with (ref overMemLimit) {
+        on loc {
+            if !localeMemAvailable(reqMemory) {
+                overMemLimit = true;
+            }
+        }
+      }
+      
+      return if overMemLimit then false else true;
+    }
+
+    proc main() {
+      try {
+          writeln("is mem available: %t".format(isMemAvailable(10000000)));
+      } catch e: Error{
+          try! writeln("error: %t".format(e));
+      }
+   }
```

Comment:
this looks like it was used for testing while running this as a stand alone. Do we still want this?

Reply:
@pierce314159 good point, gonna rip it out now

Reply:
done